### PR TITLE
 do not expand ctes when the user supplies a cte 

### LINKED ISSUE #753: error when running CTEs
since we automatically resolve dependencies, a CTE will break if one of them contains a name that's been used already by a given snippet:

```
In [1]: %load_ext sql

In [2]: %sql duckdb:// --alias myconnection

In [3]: %sql --save penguins select * from penguins.csv
Running query in 'myconnection'
Out[3]:
+---------+-----------+----------------+---------------+-------------------+-------------+--------+
| species |   island  | bill_length_mm | bill_depth_mm | flipper_length_mm | body_mass_g |  sex   |
+---------+-----------+----------------+---------------+-------------------+-------------+--------+
|  Adelie | Torgersen |      39.1      |      18.7     |        181        |     3750    |  MALE  |
|  Adelie | Torgersen |      39.5      |      17.4     |        186        |     3800    | FEMALE |
|  Adelie | Torgersen |      40.3      |      18.0     |        195        |     3250    | FEMALE |
|  Adelie | Torgersen |      None      |      None     |        None       |     None    |  None  |
|  Adelie | Torgersen |      36.7      |      19.3     |        193        |     3450    | FEMALE |
|  Adelie | Torgersen |      39.3      |      20.6     |        190        |     3650    |  MALE  |
|  Adelie | Torgersen |      38.9      |      17.8     |        181        |     3625    | FEMALE |
|  Adelie | Torgersen |      39.2      |      19.6     |        195        |     4675    |  MALE  |
|  Adelie | Torgersen |      34.1      |      18.1     |        193        |     3475    |  None  |
|  Adelie | Torgersen |      42.0      |      20.2     |        190        |     4250    |  None  |
+---------+-----------+----------------+---------------+-------------------+-------------+--------+

In [4]: %sql select * from penguins
Generating CTE with stored snippets: 'penguins'
Running query in 'myconnection'
Out[4]:
+---------+-----------+----------------+---------------+-------------------+-------------+--------+
| species |   island  | bill_length_mm | bill_depth_mm | flipper_length_mm | body_mass_g |  sex   |
+---------+-----------+----------------+---------------+-------------------+-------------+--------+
|  Adelie | Torgersen |      39.1      |      18.7     |        181        |     3750    |  MALE  |
|  Adelie | Torgersen |      39.5      |      17.4     |        186        |     3800    | FEMALE |
|  Adelie | Torgersen |      40.3      |      18.0     |        195        |     3250    | FEMALE |
|  Adelie | Torgersen |      None      |      None     |        None       |     None    |  None  |
|  Adelie | Torgersen |      36.7      |      19.3     |        193        |     3450    | FEMALE |
|  Adelie | Torgersen |      39.3      |      20.6     |        190        |     3650    |  MALE  |
|  Adelie | Torgersen |      38.9      |      17.8     |        181        |     3625    | FEMALE |
|  Adelie | Torgersen |      39.2      |      19.6     |        195        |     4675    |  MALE  |
|  Adelie | Torgersen |      34.1      |      18.1     |        193        |     3475    |  None  |
|  Adelie | Torgersen |      42.0      |      20.2     |        190        |     4250    |  None  |
+---------+-----------+----------------+---------------+-------------------+-------------+--------+

In [5]: %%sql
   ...: with something as(
   ...: select * from penguins
   ...: )
   ...: select * from something
   ...:
   ...:
Generating CTE with stored snippets: 'penguins'
Running query in 'myconnection'
RuntimeError: If using snippets, you may pass the --with argument explicitly.
For more details please refer : https://jupysql.ploomber.io/en/latest/compose.html#with-argument


Original error message from DB driver:
(duckdb.ParserException) Parser Error: syntax error at or near "with"
LINE 2: with something as(
        ^
[SQL: WITH penguins AS (select * from penguins.csv)
with something as(
select * from penguins
)
select * from something]
(Background on this error at: https://sqlalche.me/e/14/f405)

If you need help solving this issue, send us a message: https://ploomber.io/community
```

### PR DESCRIPTION
**Pull request by edublancas**
## Issue number

closes #753

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--795.org.readthedocs.build/en/795/

### CODE REVIEW

File: src/tests/test_magic.py [hunk 1]
```diff
@@ -1664,3 +1665,35 @@ def test_warning_if_variable_defined_but_named_param_is_quoted(
         match=expected_warning,
     ):
         ip.run_cell(cell)
+
+
+def test_run_cte(ip):
```
Comment (neelasha23) [inline src/tests/test_magic.py:1670]: I think we can add a comment here why this test case is important (to handle case when query is itself a cte)

**Comment by neelasha23** [comment]
Still getting this error:
Screenshot 2023-08-11 at 10 37 14 PM

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## 0.9.2dev
 
+* [Fix] Fix a bug that caused a CTE to fail if there was a snippet defined with the same name (#753)
```
Comment (neelasha23) [inline CHANGELOG.md:5]: `same name as table`

**Comment by neelasha23** [comment]
It looks like, if there's a table `some-table` and we create a snippet `some-table` and run the below query:
```
%%sql
with something as(
select * from some-table
)
select * from something
```

this query refers to the main table `some-table`. What if user wants to refer to the saved snippet instead? If I pass `--with` it throws error (there's a table as well as snippet by name `names2`):

Screenshot 2023-08-11 at 10 51 36 PM

File: CHANGELOG.md (hunk 2 above)
Comment (edublancas) [reply CHANGELOG.md]: [reworded](https://github.com/ploomber/jupysql/pull/795/commits/c97717619dd7cb00ff226eac64c7f7d10c31e625)

File: src/tests/test_magic.py (hunk 1 above)
Comment (edublancas) [reply src/tests/test_magic.py]: added a clearer [test name](https://github.com/ploomber/jupysql/pull/795/commits/7c749e9d9ce8556fce4ef21c98e76c13b4b7e219)

**Comment by edublancas** [comment]
@neelasha23 I'm now raising an exception to tell the user they should not use `--with` if they're running a CTE: https://github.com/ploomber/jupysql/pull/795/commits/143edc15346a1d3ba9fc9cfac995688fbf6ee28c

**Review by neelasha23 — APPROVED** [review]

**Pull request merged**